colossalai/zero/shard_utils/base_shard_strategy.py

```diff
@@ -13,14 +27,14 @@ def __init__(self, process_group: Optional[dist.ProcessGroup] = None) -> None:
         super().__init__()
 
     @abstractmethod
-    def shard(self, tensor_list: List[ShardedTensor]):
+    def shard(self, tensor_list: List[Tensor]) -> List[Tensor]:
```

Comment:
maybe we should change the payload of tensors in the tensor_list inline.
the function should not return a value?
The same as gather?

Reply:
I think this depends on the sharding granularity. If we shard a tensor, the sharded tensor will do. If we shard a tensor list, then we cannot store the sharded output into the payload.

Reply:
I see. 
If so, the input tensor shall not be changed.
We can unify the design whether shard() can make impact on input tensors.